### PR TITLE
Add python windows #67

### LINKED ISSUE #67: Support Python on Windows
I'd like to use pyroscope for Python on Windows.

### PR DESCRIPTION
**Pull request by mjmdavis**
closes #67 
I am looking into how to add support for Python on Windows.

The main issue seems to be a platform dependent type in a transitive dependency of `pyroscope_pyspy`. I'm not sure what the best method for addressing this is but I went with adding a direct dependency and pulling the type in from there.

### CODE REVIEW

File: pyroscope_backends/pyroscope_pyspy/src/lib.rs [hunk 1]
```diff
@@ -29,7 +30,7 @@ pub fn pyspy_backend(config: PyspyConfig) -> BackendImpl<BackendUninitialized> {
 #[derive(Debug, Clone)]
 pub struct PyspyConfig {
     /// Process to monitor
-    pid: Option<i32>,
+    pid: Option<Pid>,
```
Comment (mjmdavis) [inline pyroscope_backends/pyroscope_pyspy/src/lib.rs:33]: Use the type `remoteprocess::Pid` that is dynamic depending on the target platform.

File: pyroscope_ffi/ffikit/Cargo.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@ edition = "2021"
 
 [dependencies]
 interprocess = { version = "1.1.1", features = ["nonblocking"] }
-pyroscope = "0.5.3"
+pyroscope = { path  = "../../" }
```
Comment (mjmdavis) [inline pyroscope_ffi/ffikit/Cargo.toml:10]: Ensure we're referencing the checked out version of pyroscope-rs.

File: src/utils.rs [hunk 3]
```diff
@@ -74,11 +74,19 @@ mod check_err_tests {
 }
 
 /// libc::epoll_ctl wrapper
+#[cfg(not(target_os = "windows"))]
```
Comment (mjmdavis) [inline src/utils.rs:77]: If we're not on windows, don't do anything different.

File: src/utils.rs [hunk 4]
```diff
@@ -74,11 +74,19 @@ mod check_err_tests {
...
 pub fn pthread_self() -> Result<u64> {
     let thread_id = check_err(unsafe { libc::pthread_self() })? as u64;
     Ok(thread_id)
 }
 
+
+#[cfg(target_os = "windows")]
```
Comment (mjmdavis) [inline src/utils.rs:84]: if we are on windows, use `winapi` to get the thread id.

**Comment by mjmdavis** [comment]
I have the windows build going now. I need to try installing it and see if it works.

**Comment by mjmdavis** [comment]
So, it's working on windows. I am able to send telemetry to https://pyroscope.cloud and everything :)

**Comment by korniltsev** [comment]
Hi. Thanks a lot for contribution and sorry for long response.
I'l merge it now.
I am probably going to remove the ci-ffi-python-windows.yml and keep the ci-ffi-python.yml. Please let me know if I misunderstood the intent of the windows file.
I am also going to add it to the publish gh action.

Thanks a lot!

**Pull request merged**